src/views/virtualmachines/tree/VirtualMachineTreeView.tsx

```diff
@@ -0,0 +1,136 @@
+import { useSyncClicksEffects } from './hooks/useSyncClicksEffects';
+import { useTreeViewData } from './hooks/useTreeViewData';
+import { useTreeViewSearch } from './hooks/useTreeViewSearch';
+import {
+  FOLDER_SELECTOR_PREFIX,
+  PROJECT_SELECTOR_PREFIX,
+  TREE_VIEW_PANEL_ID,
+  VM_FOLDER_LABEL,
+} from './utils/constants';
+import {
+  createTreeViewData,
+  selectedTreeItem,
+  setSelectedTreeItem,
+  treeViewOpen,
+} from './utils/utils';
+
+import './VirtualMachineTreeView.scss';
+
+type VirtualMachineTreeViewProps = {
+  onFilterChange?: (type: string, value: FilterValue) => void;
+};
+
+const VirtualMachineTreeView: FC<VirtualMachineTreeViewProps> = ({ children, onFilterChange }) => {
+  const [activeNamespace] = useActiveNamespace();
+  const navigate = useNavigate();
+  const location = useLocation();
+
+  const { setOrRemoveQueryArgument } = useQueryParamsMethods();
+  const { isAdmin, loaded, loadError, projectNames, vms } = useTreeViewData();
+  const vmsMapper = useMemo(() => convertResourceArrayToMap(vms, true), [vms]);
+
+  const treeData = useMemo(
+    () => createTreeViewData(projectNames, vms, activeNamespace, isAdmin, location.pathname),
+    [projectNames, vms, activeNamespace, isAdmin, location.pathname],
+  );
+
+  const { filteredItems, onSearch, setShowAll, showAll } = useTreeViewSearch(treeData);
+
+  useSyncClicksEffects(activeNamespace, loaded, location);
+  useHideNamespaceBar();
+
+  if (loadError) return <>{children}</>;
+
+  const onSelect = (_event: MouseEvent, treeViewItem: TreeViewDataItem) => {
+    setSelectedTreeItem(treeViewItem);
+    onFilterChange?.(TEXT_FILTER_LABELS_ID, null);
+
+    const treeItemName = treeViewItem.name as string;
+    if (treeViewItem.id.startsWith(FOLDER_SELECTOR_PREFIX)) {
+      const [_, folderNamespace] = treeViewItem.id.split('/');
+      navigate(
+        getResourceUrl({
+          activeNamespace: folderNamespace,
+          model: VirtualMachineModel,
+        }),
+      );
+      setOrRemoveQueryArgument(TEXT_FILTER_LABELS_ID, `${VM_FOLDER_LABEL}=${treeItemName}`);
+      return onFilterChange?.(TEXT_FILTER_LABELS_ID, {
+        all: [`${VM_FOLDER_LABEL}=${treeItemName}`],
+      });
+    }
+
+    if (treeViewItem.id.startsWith(PROJECT_SELECTOR_PREFIX)) {
+      return navigate(
+        getResourceUrl({
+          activeNamespace: treeItemName,
+          model: VirtualMachineModel,
+        }),
+      );
+    }
+
+    const [vmNamespace, vmName] = treeViewItem.id.split('/');
+    return navigate(
+      getResourceUrl({
+        activeNamespace: vmNamespace,
+        model: VirtualMachineModel,
+        resource: vmsMapper?.[vmNamespace]?.[vmName],
+      }),
+    );
+  };
+
+  return (
+    <Drawer isExpanded isInline position="start">
+      <DrawerContent
+        panelContent={
+          <DrawerPanelContent
+            style={{
+              height: document.getElementById('content-scrollable').offsetHeight || 0,
+              overflowX: 'hidden',
```

Comment:
u can spread a class here, i understand the issue with height, but other values can come from class